exist-core/src/test/xquery/unicode.xqm

```diff
@@ -70,6 +70,14 @@ function uni:sip-dom-persistent() {
         fn:string-to-codepoints(fn:doc($stored)/container/smp)
 };
 
+declare
+    %test:assertEquals(172998)
+function uni:sip-dom-persistent-text() {
+    let $stored := xmldb:store("/db/" || $uni:COLLECTION_NAME, "sip.xml", $uni:SIP)
+    return
+        fn:string-to-codepoints(fn:doc($stored/text())/container/smp)
```

Comment:
```suggestion
        fn:string-to-codepoints(fn:doc($stored)/container/smp)/text()
```

i think there is a syntax problem

Reply:
Strictly speaking, the `text()` node test is not necessary, as the string value of the `<smp>` element will be atomized when it's passed to `fn:string-to-codepoints`. It's harmless, so unless this node test is critical to demonstrating a failure (and it's not, according to the results in @wolfgangmm and my tests above), it should be removed entirely.

Reply:
yes the test doesn't seem to reproduce the original issue, I ll try to create tweak it to reproduce the code in #3651. As reported there the combination of `text()` selector on a permanent node is what is bugged (at least in some circumstances?)